modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistService.java

```diff
@@ -0,0 +1,400 @@
+      if (!checkPermission(playlist, Permissions.Action.WRITE)) {
+        throw new UnauthorizedException("User does not have read permissions");
+      }
+    } catch (PlaylistDatabaseException e) {
+      throw new IllegalStateException(e);
+    }
+
+    playlist.removeEntry(
+        playlist.getEntries()
+            .stream()
+            .filter(e -> e.getId() == entryId)
+            .collect(Collectors.toList())
+            .get(0)
+    );
+
+    try {
+      playlist = persistence.updatePlaylist(playlist, securityService.getOrganization().getId());
+      return playlist;
+    } catch (PlaylistDatabaseException e) {
+      throw new IllegalStateException("Could not delete playlist from database with id ");
+    }
+  }
+
+  /**
+   * Enrich each entry of a playlist with information about the event. Intended to be used by endpoints when
+   * returning information about a playlist.
+   * @param playlist The playlist to enrich
+   * @return The serialization class of the playlist, since the added information does not belong to the playlist
+   * itself.
+   */
+  public JaxbPlaylist enrich(Playlist playlist) {
+    JaxbPlaylist jaxbPlaylist = new JaxbPlaylist(playlist);
+
+    // Add additional infos about events
+    List<JaxbPlaylistEntry> jaxbPlaylistEntries = jaxbPlaylist.getEntries();
+    for (int i = 0; i < jaxbPlaylistEntries.size(); i++) {
+      try {
+        JaxbPlaylistEntry entry = jaxbPlaylistEntries.get(i);
+        Opt<Event> optEvent = indexService.getEvent(jaxbPlaylistEntries.get(i).getEventId(), elasticsearchIndex);
+        // We only get an event from the indexService if we have permission to do so (and if it exists ofc)
+        if (optEvent.isSome()) {
+          Event event = optEvent.get();
+          entry.setPublications(event.getPublications());
+        } else {
+          entry.setType(PlaylistEntryType.INACCESSIBLE);
+        }
+        jaxbPlaylistEntries.set(i, entry);
+      } catch (SearchIndexException e) {
+        throw new RuntimeException(e);
+      }
+    }
+    jaxbPlaylist.setEntries(jaxbPlaylistEntries);
+
+    return jaxbPlaylist;
+  }
+
+  /**
+   * Runs a permission check on the given playlist for the given action
+   * @param playlist {@link Playlist} to check permission for
+   * @param action Action to check permission for
+   * @return True if action is permitted on the {@link Playlist}, else false
+   */
+  private boolean checkPermission(Playlist playlist, Permissions.Action action) {
+    User currentUser = securityService.getUser();
+    Organization currentOrg = securityService.getOrganization();
+    String currentOrgAdminRole = currentOrg.getAdminRole();
+    String currentOrgId = currentOrg.getId();
+
+    return currentUser.hasRole(GLOBAL_ADMIN_ROLE)
+        || (currentUser.hasRole(currentOrgAdminRole) && currentOrgId.equals(playlist.getOrganization()))
+        || authorizationService.hasPermission(getAccessControlList(playlist), action.toString());
+  }
```

Comment:
@Arnei does this mean the Playlist ACL is independent of the the Mediapackage ACL? What happens when the player tries to play the media within a playlist and doesn't have access to the actual media?

Reply:
Correct, the Playlist ACL is independent from any ACL its entries may have. The Playlist ACL should determine what a user can do with the playlist, not with its entries. So even if the player has read rights on a given playlist, media may still be inaccessible if the player not also has read rights for the events, based on the respective mediapackage ACLs.

Reply:
Might be more helpful if I link the spec I got for this: https://hedgedoc.uni-osnabrueck.de/VxvRJXbWT7SlZGUtMJVHcg?view#